src/node.h

```diff
@@ -487,7 +487,7 @@ NODE_DEPRECATED("Use WinapiErrnoException(isolate, ...)",
 #endif
 
 const char* signo_string(int errorno);
-
+const std::string GetOpenSSLVersion();
```

Comment:
I'd rather not declare this as a public API.
IMHO no declaration is required, just put in higher in `node.cc`, but if you feel a decl is required, I think `util.h` is a better place.

Reply:
If this is split out into a separate function, it should be defined in `node_crypto.h`, as is all other OpenSSL-specific code.

Reply:
It could be useful in the node-report PR, especially if we go down the JSON format route (where I think this sliced version string is more appropriate than the full string currently used). But declaring the function (in `node_crypto.h`) could easily be done over in that PR.

Reply:
@richardlau Regarding the node-report PR … it still wouldn’t have to be *public* API, right?

Reply:
> Regarding the node-report PR … it still wouldn’t have to be public API, right?

Correct. It doesn't need to be declared at all in this PR (as it's only used in `src/node.cc`). For use in the node-report PR it should be declared as a node internal.

Reply:
followed the suggestions; moved things to `node-crypto`.

to be clear: what is a `public API` in this context? symbols accessible outside of `node.exe`? methods visible and callable when node is a shared library? AFAIK visibility of symbols in object files are platform dependent, and are managed (for embedders) manually at the moment? In the absence of that, every method is visible in say Linux? or is it that we follow a convention that all declared methods in `node.h` is deemed `public`? please let me know. thanks!

Reply:
> what is a `public API` in this context

Methods in headers that are included in the header tarball and not guarded by `NODE_WANT_INTERNALS`, i.e.:

    #include <node.h>
    #include <node_api.h>
    #include <node_buffer.h>
    #include <node_version.h>
    #include <node_object_wrap.h>